# Esercizio1

Si consideri il dataset delle piogge mensili (mm) nei sette anni dal 2009 al 2015, che ha il seguente formato: 13 record di campi separati da tabulazione di cui il primo è il record di intestazione degli anni (composto da 7 campi) e gli altri 12 sono i record delle piogge mensili (un record per ciascun mese).
Ogni record è composto di 8 campi, di cui il primo è il nome del mese e i rimanenti 7 sono le piogge mensili lungo gli anni.

Si richiede di determinare:

- per ogni mese, la pioggia media lungo gli anni
- per ogni anno, la pioggia totale
- per ogni anno, il numero di mesi con pioggia oltre una certa soglia (parametro in input)

---

Parametri di input:
- dataset delle piogge
- soglia minima di pioggia

---

### Parametri di input

### Definizione di una funzione che conta i valori di una lista oltre una certa soglia

### Lettura del dataset

- Funzione `open(file_name, 'r')` per ottenere l'*handle* in lettura al file
- Metodo `readlines()` per leggere il canale invocante in una lista di stringhe
- Metodo `close()` per chiudere l'*handle* invocante

### Costruzione della lista degli anni rilevati

In [84]:
years

['2009', '2010', '2011', '2012', '2013', '2014', '2015']

### Costruzione della lista dei mesi

In [87]:
months

['Gennaio',
 'Febbraio',
 'Marzo',
 'Aprile',
 'Maggio',
 'Giugno',
 'Luglio',
 'Agosto',
 'Settembre',
 'Ottobre',
 'Novembre',
 'Dicembre']

### Costruzione della matrice delle piogge

### Trasposta della matrice delle piogge

### Piogge medie mensili

#### a) Costruzione della lista delle piogge medie mensili

In [97]:
avg_rain_list

[66.42857142857143,
 60.42857142857143,
 83.28571428571429,
 88.28571428571429,
 123.42857142857143,
 110.28571428571429,
 105.85714285714286,
 126.85714285714286,
 91.14285714285714,
 108.42857142857143,
 111.0,
 58.285714285714285]

#### b) Output delle piogge medie mensili

Produrre il seguente output (stdout)

    GEN -> 66.42857142857143
    FEB -> 60.42857142857143
    MAR -> 83.28571428571429
    APR -> 88.28571428571429
    MAG -> 123.42857142857143
    GIU -> 110.28571428571429
    LUG -> 105.85714285714286
    AGO -> 126.85714285714286
    SET -> 91.14285714285714
    OTT -> 108.42857142857143
    NOV -> 111.0
    DIC -> 58.285714285714285

### Piogge totali annue

#### a) Costruzione della lista delle piogge totali annue

In [102]:
year_total_rain

[1148, 1154, 1084, 1113, 1169, 1135, 1133]

---

#### b) Ouput delle piogge totali annue

Produrre il seguente output (stdout)

    2009 -> 1148
    2010 -> 1154
    2011 -> 1084
    2012 -> 1113
    2013 -> 1169
    2014 -> 1135
    2015 -> 1133

### Numero di mesi per anno con pioggia oltre la soglia minima

#### a) Costruzione della lista del numero di mesi (per anno) con pioggia oltre la soglia minima

In [116]:
year_count

[6, 5, 6, 6, 5, 6, 4]

#### b) Output del numero di mesi (per anno) con pioggia oltre la soglia minima

Produrre il seguente output (stdout)

    2009 -> 6
    2010 -> 5
    2011 -> 6
    2012 -> 6
    2013 -> 5
    2014 -> 6
    2015 -> 4